lua/markview/integrations.lua

```diff
@@ -17,6 +17,43 @@ integrations.register_blink_source = function ()
 		ignore_enable = true
 	});
 
+	--- Dynamic filetype.
+	---@param filetype string
+	---@param config table | fun(): table See |blink-cmp-config-sources|.
+	local function handle_filetype(filetype, config)
+		---|fS
+
+		if type(config) == "table" and vim.islist(config) then
+			---@cast config table
+			if vim.list_contains(config, "markview") then
+				-- `markview` is already used as a source.
+				return;
+			end
+
+			-- If the config sources is a list we extend the list of sources.
+			blink_config.sources.per_filetype[filetype] = vim.list_extend(config, { "markview" });
```

Comment:
This is a mistake I also made in my original implementation, and just caught it now:

`vim.list_extend` modifies the list in place. Since this `if` branch essentially asserts that we're dealing with the `default` sources key, we're essentially appending the `markview` source to the default list, and copying  that list to the `per_filetype.markdown` sources, which is incorrect behavior. This means the `markview` source will be loaded on any filetype, not just markdowns.